mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -1297,28 +1331,44 @@ void DeleteMember (string reason, string output, XmlNode member, MyXmlNodeList t
 				member.OwnerDocument.DocumentElement.GetAttribute ("FullName"),
 				member.Attributes ["MemberName"].Value, 
 				member.SelectSingleNode ("MemberSignature[@Language='C#']/@Value").Value);
-			if (!delete && MemberDocsHaveUserContent (member)) {
-				Warning ("Member deletions must be enabled with the --delete option.");
-			} else if (HasDroppedNamespace (type)) {
-				// if we're dropping the namespace, add the "classic style"
-				var existingAttribute = member.Attributes ["apistyle"];
-				if (existingAttribute != null) {
-					existingAttribute.Value = "classic";
-				} else {
-					// add the attribute and do not remove
-					XmlAttribute apistyleAttr = member.OwnerDocument.CreateAttribute ("apistyle");
 
-					apistyleAttr.Value = "classic";
+		// Identify all of the different states that could affect our decision to delete the member
+		bool shouldPreserve = !string.IsNullOrWhiteSpace (PreserveTag);
+		bool hasContent = MemberDocsHaveUserContent (member);
+		bool shouldDelete = !shouldPreserve && ((delete && !hasContent) || (!delete && !hasContent));
```

Comment:
...I hate boolean logic.

`((delete && !hasContent) || (!delete && !hasContent))` should be `!hasContent`, no?

<table>
  <tr><th></th><th>delete=true</th><th>delete=false</th></tr>
  <tr><th>hasContent=true</th><td>false</td><td>false</td></tr>
  <tr><th>hasContent=false</th><td>true</td><td>true</td></tr>  
</table>

Assuming my truth table is right (...), the only possible outcome of `((delete && !hasContent) || (!delete && !hasContent))` is `!hasContent`, so why check `delete` at all?